index.js

```diff
@@ -57,9 +58,16 @@ Parser.prototype._transform = function (data, enc, cb) {
   }
 
   for (var i = start; i < buf.length; i++) {
-    // non-escaped quote (quoting the cell)
-    if (buf[i] === this.quote && (i === start || (i > start && buf[i - 1] !== this.escape))) {
-      this._quoted = !this._quoted
+    if (!this._escaped && buf[i] === this.escape && buf[i + 1] === this.quote && i !== start) {
```

Comment:
could you check that i + i < buf.length as well? otherwise we're gonna get a deopt from v8